### PR TITLE
style(frontend): implement color scheme and theme from figma

### PR DESCRIPTION
**Pull request by loki344**
# Motivation

We want to align color usages to the defined color variables from the design team.

Primitive colors:
image

Variables:
image

This PR consolidates ideas from: https://github.com/dfinity/oisy-wallet/pull/2897 and https://github.com/dfinity/oisy-wallet/pull/3121

In a nutshell:
We define available colors in base-colors.scss. (Primitive colors)
We define a theme by specifying the theme variables and assigning colors.
We make the variables available in tailwind.

# Note
This PR showcases usages for brand colors. If approved, the next PR will implement all colors. And then a further PR will replace current color usages with the new scheme.

# Changes

- base-colors.scss introducing a place to define primitive colors
- default-theme.scss using these colors to define a theme
- theme-variables.ts to make the theme variables available in tailwind

# Tests

A quick showcasing of the application:
https://github.com/user-attachments/assets/e7d5f707-2b28-44a9-8377-17c7259721af

Example:
Overriding the theme will be:
1) define a new theme.scss with a new color mapping
2) make the :root selector more specific, switching to [data-theme='fancy']

For showcasing i changed the brand color to greenish.
https://github.com/user-attachments/assets/67dc121d-5b6c-47e8-8585-5928dc64c535

### CODE REVIEW

**Comment by loki344** [comment]
@AntonioVentilii-DFINITY  / @peterpeterparker  let me know what you think.

**Comment by AntonioVentilii-DFINITY** [comment]
@loki344 As discussed offline, better to start with the "brand" colors:

- the Design team will adjust the neutrals colors
- the brand colors give more practical examples on how the color schema is applied

**Comment by loki344** [comment]
@AntonioVentilii-DFINITY take a look at it again please

File: src/frontend/src/lib/components/tokens/TokensZeroBalance.svelte [hunk 1]
```diff
@@ -4,7 +4,7 @@
 </script>
 
 <div class="content">
-	<span class="block text-sm text-blue-ribbon">{$i18n.tokens.text.balance}</span>
+	<span class="block text-sm text-foreground-brand-primary">test{$i18n.tokens.text.balance}</span>
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/components/tokens/TokensZeroBalance.svelte:7]: here there is an intruder? eheh string `test` ?

File: src/frontend/src/lib/styles/global/button.scss
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/styles/global/button.scss:1]: for this one, can you just do me a favour? run locally `npm run build` and check if SASS is giving any warnings?

File: src/frontend/src/lib/styles/theme/base-colors.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+$alice-blue: #e8f1ff;
+$water: #d1e3ff;
+$pale-cornflower-blue: #b0cdff;
+$french-sky-blue: #84b2ff;
+$crayola: #004abe;
+$catalina-blue: #012f80;
+$brandeis-blue: #0066ff;
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/styles/theme/base-colors.scss:7]: ah! is it not `blue-ribbon`? I swear I thought it was

File: src/frontend/src/lib/styles/theme/base-colors.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+$alice-blue: #e8f1ff;
+$water: #d1e3ff;
+$pale-cornflower-blue: #b0cdff;
+$french-sky-blue: #84b2ff;
+$crayola: #004abe;
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/lib/styles/theme/base-colors.scss:5]: I think this is quite generic for a color name, we were calling it `cobalt`, no?

Comment (loki344) [reply src/frontend/src/lib/styles/theme/base-colors.scss:5]: Yeah I can change it, i used 	// https://www.color-name.com which is mentioned in colors.scss - I have no idea why it gave me different names xD

File: src/frontend/src/lib/styles/theme/base-colors.scss (hunk 2 above)
Comment (loki344) [reply src/frontend/src/lib/styles/theme/base-colors.scss:7]: I'm using whatever pops out here :D
image

File: src/frontend/src/lib/components/tokens/TokensZeroBalance.svelte (hunk 1 above)
Comment (loki344) [reply src/frontend/src/lib/components/tokens/TokensZeroBalance.svelte]: thanks for spotting this

File: src/frontend/src/lib/styles/global/button.scss
Comment (loki344) [reply src/frontend/src/lib/styles/global/button.scss:1]: We're clear

`robertschlittler@MacBook-Pro-von-Robert oisy-wallet % npm run build

> @dfinity/oisy-wallet@0.9.2 build
> tsc --noEmit && vite build && npm run build:post-process

vite v5.4.9 building SSR bundle for production...
✓ 1058 modules transformed.
vite v5.4.9 building for production...
✓ 1669 modules transformed.
.svelte-kit/output/client/_app/version.json                                                   0.02 kB │ gzip:   0.04 kB
.svelte-kit/output/client/_app/immutable/assets/uni.UZkkJLlf.svg                              4.72 kB │ gzip:   2.17 kB
.svelte-kit/output/client/_app/immutable/workers/assets/uni-UZkkJLlf.svg                      4.72 kB │ gzip:   2.17 kB
.svelte-kit/output/client/_app/immutable/assets/oisy-logo-large.Dyk8dCuj.svg                  4.78 kB │ gzip:   1.89 kB
.svelte-kit/output/client/_app/immutable/assets/icp_dark.CwKXLwSR.svg                         5.58 kB │ gzip:   2.00 kB
.svelte-kit/output/client/_app/immutable/assets/1inch.DRmAKuj0.svg                            6.34 kB │ gzip:   2.42 kB
.svelte-kit/output/client/_app/immutable/assets/banner.CZmHnsB8.svg                           6.56 kB │ gzip:   2.23 kB
.svelte-kit/output/client/.vite/manifest.json                                                 8.50 kB │ gzip:   1.13 kB
.svelte-kit/output/client/_app/immutable/assets/shib.Bgq9R77i.svg                             9.35 kB │ gzip:   3.82 kB
.svelte-kit/output/client/_app/immutable/workers/assets/shib-Bgq9R77i.svg                     9.35 kB │ gzip:   3.82 kB
.svelte-kit/output/client/_app/immutable/assets/jasmy.7baMhIky.svg                           10.35 kB │ gzip:   7.44 kB
.svelte-kit/output/client/_app/immutable/assets/cover-why-oisy.DddJE7Zq.png                  17.56 kB
.svelte-kit/output/client/_app/immutable/assets/pepe.Dqx2_DUA.svg                            18.23 kB │ gzip:   8.46 kB
.svelte-kit/output/client/_app/immutable/workers/assets/pepe-Dqx2_DUA.svg                    18.23 kB │ gzip:   8.46 kB
.svelte-kit/output/client/_app/immutable/assets/dmail.C2AFR0Uq.svg                           24.60 kB │ gzip:  18.63 kB
.svelte-kit/output/client/_app/immutable/assets/floki.DJjLPgkR.svg                           38.14 kB │ gzip:  27.73 kB
.svelte-kit/output/client/_app/immutable/assets/weeth.614_0O9m.svg                           43.53 kB │ gzip:  32.94 kB
.svelte-kit/output/client/_app/immutable/assets/preview.DNKZ_gZC.webp                        55.92 kB
.svelte-kit/output/client/_app/immutable/workers/exchange.worker-DghpXpLg.js                178.55 kB
.svelte-kit/output/client/_app/immutable/workers/auth.worker-Cs83LtQI.js                    224.89 kB
.svelte-kit/output/client/_app/immutable/workers/btc-wallet.worker-BsPnOHhc.js              353.08 kB
.svelte-kit/output/client/_app/immutable/workers/cketh-minter-info.worker-Bljrjgpl.js       356.93 kB
.svelte-kit/output/client/_app/immutable/workers/ckbtc-minter-info.worker-Ck8UV0Y5.js       361.98 kB
.svelte-kit/output/client/_app/immutable/workers/btc-statuses.worker-B9bR3RtN.js            362.00 kB
.svelte-kit/output/client/_app/immutable/workers/icp-wallet.worker-EM37kZT6.js              383.82 kB
.svelte-kit/output/client/_app/immutable/workers/icrc-wallet.worker-5TuzqUu9.js             504.99 kB
.svelte-kit/output/client/_app/immutable/workers/ckbtc-update-balance.worker-CPJ19c4R.js    510.50 kB
.svelte-kit/output/client/_app/immutable/assets/index.a88HJGRz.css                          106.80 kB │ gzip:  20.09 kB
.svelte-kit/output/client/_app/immutable/entry/start.Emx6UT7o.js                              0.07 kB │ gzip:   0.08 kB
.svelte-kit/output/client/_app/immutable/nodes/1.DkCf2PoH.js                                  0.07 kB │ gzip:   0.09 kB
.svelte-kit/output/client/_app/immutable/nodes/2.Gr3zn47P.js                                  0.07 kB │ gzip:   0.09 kB
.svelte-kit/output/client/_app/immutable/nodes/3.yPovziqE.js                                  0.07 kB │ gzip:   0.09 kB
.svelte-kit/output/client/_app/immutable/nodes/4.BsEYDJ0U.js                                  0.07 kB │ gzip:   0.09 kB
.svelte-kit/output/client/_app/immutable/nodes/5.CVirAapr.js                                  0.07 kB │ gzip:   0.09 kB
.svelte-kit/output/client/_app/immutable/nodes/0.xywcKeOQ.js                                  0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/nodes/6.Dod8QMgP.js                                  0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/nodes/7.DlioivS0.js                                  0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/nodes/8.DWUy1Wmn.js                                  0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/nodes/9.Ciup168T.js                                  0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/nodes/10.YuPtzHiQ.js                                 0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/nodes/11.DmMgv1jA.js                                 0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/nodes/12.Dwwi9DNq.js                                 0.09 kB │ gzip:   0.10 kB
.svelte-kit/output/client/_app/immutable/entry/app.CvCuLqo3.js                                0.17 kB │ gzip:   0.15 kB
.svelte-kit/output/client/_app/immutable/chunks/lazy.Daz1gh3a.js                            347.14 kB │ gzip: 105.44 kB
.svelte-kit/output/client/_app/immutable/chunks/index.DBWrDzSh.js                         1,254.74 kB │ gzip: 339.96 kB
.svelte-kit/output/client/_app/immutable/chunks/vendor.BP3Cn98K.js                        1,606.03 kB │ gzip: 508.62 kB

(!) Some chunks are larger than 500 kB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 16.74s
.svelte-kit/output/server/_app/immutable/workers/auth.worker-DAR9Ycce.js                      2.52 kB
.svelte-kit/output/server/_app/immutable/workers/exchange.worker-YXEYLD-o.js                  4.43 kB
.svelte-kit/output/server/_app/immutable/assets/uni.UZkkJLlf.svg                              4.72 kB
.svelte-kit/output/server/_app/immutable/workers/assets/uni-UZkkJLlf.svg                      4.72 kB
.svelte-kit/output/server/_app/immutable/assets/oisy-logo-large.Dyk8dCuj.svg                  4.78 kB
.svelte-kit/output/server/_app/immutable/assets/icp_dark.CwKXLwSR.svg                         5.58 kB
.svelte-kit/output/server/_app/immutable/assets/1inch.DRmAKuj0.svg                            6.34 kB
.svelte-kit/output/server/_app/immutable/assets/banner.CZmHnsB8.svg                           6.56 kB
.svelte-kit/output/server/_app/immutable/workers/ckbtc-minter-info.worker-23BAxuh0.js         6.77 kB
.svelte-kit/output/server/_app/immutable/workers/cketh-minter-info.worker-D1xQMQ9e.js         6.78 kB
.svelte-kit/output/server/_app/immutable/workers/btc-statuses.worker-B00h-w-s.js              6.89 kB
.svelte-kit/output/server/_app/immutable/assets/shib.Bgq9R77i.svg                             9.35 kB
.svelte-kit/output/server/_app/immutable/workers/assets/shib-Bgq9R77i.svg                     9.35 kB
.svelte-kit/output/server/.vite/manifest.json                                                10.03 kB
.svelte-kit/output/server/_app/immutable/assets/jasmy.7baMhIky.svg                           10.35 kB
.svelte-kit/output/server/_app/immutable/workers/icp-wallet.worker-BHvkc-yo.js               13.90 kB
.svelte-kit/output/server/_app/immutable/assets/cover-why-oisy.DddJE7Zq.png                  17.56 kB
.svelte-kit/output/server/_app/immutable/assets/pepe.Dqx2_DUA.svg                            18.23 kB
.svelte-kit/output/server/_app/immutable/workers/assets/pepe-Dqx2_DUA.svg                    18.23 kB
.svelte-kit/output/server/_app/immutable/assets/dmail.C2AFR0Uq.svg                           24.60 kB
.svelte-kit/output/server/_app/immutable/workers/btc-wallet.worker-E4jVvxh1.js               33.96 kB
.svelte-kit/output/server/_app/immutable/assets/floki.DJjLPgkR.svg                           38.14 kB
.svelte-kit/output/server/_app/immutable/assets/weeth.614_0O9m.svg                           43.53 kB
.svelte-kit/output/server/_app/immutable/assets/preview.DNKZ_gZC.webp                        55.92 kB
.svelte-kit/output/server/_app/immutable/workers/ckbtc-update-balance.worker-DeIpaJKz.js     64.05 kB
.svelte-kit/output/server/_app/immutable/workers/icrc-wallet.worker-GFJoflyM.js              78.01 kB
.svelte-kit/output/server/_app/immutable/assets/index.BE5DC0t_.css                          106.76 kB
.svelte-kit/output/server/index.js                                                            0.07 kB
.svelte-kit/output/server/entries/fallbacks/error.svelte.js                                   0.07 kB
.svelte-kit/output/server/entries/pages/(app)/_page.ts.js                                     0.07 kB
.svelte-kit/output/server/entries/pages/(public)/_layout.svelte.js                            0.07 kB
.svelte-kit/output/server/entries/pages/(app)/_page.svelte.js                                 0.07 kB
.svelte-kit/output/server/entries/pages/(app)/explore/_page.ts.js                             0.07 kB
.svelte-kit/output/server/entries/pages/(app)/settings/_page.ts.js                            0.07 kB
.svelte-kit/output/server/entries/pages/(app)/transactions/_page.ts.js                        0.07 kB
.svelte-kit/output/server/entries/pages/(app)/wc/_page.ts.js                                  0.07 kB
.svelte-kit/output/server/entries/pages/(public)/license-agreement/_page.ts.js                0.07 kB
.svelte-kit/output/server/entries/pages/(sign)/sign/_page.ts.js                               0.07 kB
.svelte-kit/output/server/entries/pages/(app)/transactions/_layout.svelte.js                  0.08 kB
.svelte-kit/output/server/entries/pages/(sign)/sign/_layout.svelte.js                         0.08 kB
.svelte-kit/output/server/entries/pages/(app)/explore/_page.svelte.js                         0.08 kB
.svelte-kit/output/server/entries/pages/(app)/settings/_page.svelte.js                        0.08 kB
.svelte-kit/output/server/entries/pages/(app)/wc/_page.svelte.js                              0.08 kB
.svelte-kit/output/server/entries/pages/(public)/license-agreement/_page.svelte.js            0.08 kB
.svelte-kit/output/server/chunks/vendor.js                                                    0.08 kB
.svelte-kit/output/server/entries/endpoints/sitemap.xml/_server.ts.js                         0.09 kB
.svelte-kit/output/server/entries/pages/(app)/_layout.svelte.js                               0.10 kB
.svelte-kit/output/server/entries/pages/(app)/transactions/_page.svelte.js                    0.10 kB
.svelte-kit/output/server/entries/pages/(sign)/sign/_page.svelte.js                           0.10 kB
.svelte-kit/output/server/entries/pages/_layout.svelte.js                                     0.12 kB
.svelte-kit/output/server/entries/pages/_layout.ts.js                                         0.14 kB
.svelte-kit/output/server/internal.js                                                         0.30 kB
.svelte-kit/output/server/chunks/index.js                                                 1,386.18 kB
✓ built in 21.91s

Run npm run preview to preview your production build locally.

> Using @sveltejs/adapter-static
Overwriting build/index.html with fallback page. Consider using a different name for the fallback.
  Wrote site to "build"
  ✔ done

> @dfinity/oisy-wallet@0.9.2 build:post-process
> npm run build:style && npm run build:metadata && npm run build:ic-domains && npm run build:ii-alternative-origins && npm run build:csp && npm run build:copy-workers && npm run build:compress


> @dfinity/oisy-wallet@0.9.2 build:style
> node scripts/build.style.mjs


> @dfinity/oisy-wallet@0.9.2 build:metadata
> node scripts/build.metadata.mjs


> @dfinity/oisy-wallet@0.9.2 build:ic-domains
> node scripts/build.ic-domains.mjs


> @dfinity/oisy-wallet@0.9.2 build:ii-alternative-origins
> node scripts/build.ii-alternative-origins.mjs


> @dfinity/oisy-wallet@0.9.2 build:csp
> node scripts/build.csp.mjs


> @dfinity/oisy-wallet@0.9.2 build:copy-workers
> node ./scripts/build.copy-workers.mjs


> @dfinity/oisy-wallet@0.9.2 build:compress
> ./scripts/build.compress.sh

robertschlittler@MacBook-Pro-von-Robert oisy-wallet % 
`

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/styles/global/button.scss:1]: thank you for that!

File: src/frontend/src/lib/styles/theme/base-colors.scss (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/styles/theme/base-colors.scss:7]: perfect then! I must have made a mistake at start eheh

**Review by AntonioVentilii-DFINITY — APPROVED** [review]

**Pull request merged**